Packages/Tracking/Physical Hands/Runtime/Scripts/PhysicalHandsButtonBase.cs

```diff
@@ -0,0 +1,407 @@
+ ******************************************************************************/
+
+
+using System;
+using System.Collections.Generic;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Leap.Unity.PhysicalHands
+{
+    [RequireComponent(typeof(Rigidbody))]
+    public class PhysicalHandsButtonBase : MonoBehaviour
+    {
+        
+        [SerializeField] 
+        protected GameObject _pressableObject;
+        [SerializeField]
+        bool _automaticTravelDistance = true;
+        [SerializeField]
+        private float _buttonTravelDistance = 0f;
+        [SerializeField]
+        bool _automaticOffsetDistance = true;
+        [SerializeField]
+        private float _buttonTravelOffset = 0.001f;
+        [SerializeField] 
+        protected bool _canBePressedByObjects = false;
+        [SerializeField] 
+        protected ChiralitySelection _whichHandCanPressButton = ChiralitySelection.BOTH;
+        [SerializeField]
+        private bool _freezeButtonTravelOnMovement = true;
+        [SerializeField]
+        private Vector3 _buttonVelocityThreshold = new Vector3(0.1f, 0.1f, 0.1f);
+
+        private const float BUTTON_PRESS_EXIT_THRESHOLD = 0.5f;
+        private bool _isButtonPressed = false;
+        private bool _contactHandPressing = false;
+        private bool _leftHandContacting = false;
+        private bool _rightHandContacting = false;
+        private Rigidbody _pressableObjectRB = null;
+        private Rigidbody _rigidbody = null;
+        private ConfigurableJoint _configurableJoint;
+        private Vector3 _initialButtonPosition = Vector3.zero;
+        private List<GameObject> _objectsContactingButton = new List<GameObject>();
+
+        protected PhysicalHandsButtonHelper _buttonHelper;
+
+        [SerializeField]
+        public UnityEvent OnButtonPressed;
+        [SerializeField]
+        public UnityEvent OnButtonUnPressed;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandContact;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandContactExit;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandHover;
+        [SerializeField]
+        public UnityEvent<ContactHand> OnHandHoverExit;
+
+#if UNITY_EDITOR
+        public void UpdateDistanceValues()
+        {
+            if (!Application.isPlaying)
+            {
+                if (_automaticOffsetDistance)
+                {
+                    if (_pressableObject != null &&
+                        _pressableObject.GetComponent<MeshFilter>() &&
+                        GetComponent<MeshFilter>())
+                    {
+                        _buttonTravelOffset = GetComponent<MeshFilter>().sharedMesh.bounds.extents.y;
+                    }
+                    else if (UltraleapSettings.Instance.showPhysicalHandsButtonOffsetWarning == true)
+                    {
+                        if (EditorUtility.DisplayDialog("Could not automatically set offset distance.",
```

Comment:
I also get this when adding the component manually to while setting up my own Button

It may be worth just logging a warning, to the console rather than causing a popup here